app/src/main/kotlin/com/wire/android/ui/home/conversationslist/common/ConversationItemFactory.kt

```diff
@@ -1,41 +1,103 @@
 package com.wire.android.ui.home.conversationslist.common
 
 import androidx.compose.runtime.Composable
-import com.wire.android.ui.home.conversationslist.CallConversationItem
-import com.wire.android.ui.home.conversationslist.GeneralConversationItem
-import com.wire.android.ui.home.conversationslist.MentionConversationItem
+import com.wire.android.ui.common.RowItemTemplate
+import com.wire.android.ui.home.conversationslist.CallLabel
+import com.wire.android.ui.home.conversationslist.ConnectionLabel
+import com.wire.android.ui.home.conversationslist.MentionLabel
+import com.wire.android.ui.home.conversationslist.MutedConversationBadge
 import com.wire.android.ui.home.conversationslist.model.ConversationItem
 import com.wire.android.ui.home.conversationslist.model.ConversationMissedCall
+import com.wire.android.ui.home.conversationslist.model.ConversationType
 import com.wire.android.ui.home.conversationslist.model.ConversationUnreadMention
 import com.wire.android.ui.home.conversationslist.model.EventType
 import com.wire.android.ui.home.conversationslist.model.GeneralConversation
+import com.wire.android.ui.home.conversationslist.model.PendingConnectionItem
+import com.wire.android.ui.home.conversationslist.model.toUserInfoLabel
+import com.wire.kalium.logic.data.conversation.MutedConversationStatus
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.data.user.toUserId
 
 @Composable
 fun ConversationItemFactory(
     conversation: ConversationItem,
     eventType: EventType? = null,
-    onConversationItemClick: (ConversationItem) -> Unit,
-    onConversationItemLongClick: (ConversationItem) -> Unit,
+    openConversation: () -> Unit,
+    onConversationItemLongClick: () -> Unit,
+    openUserProfile: (UserId) -> Unit,
+    onMutedIconClick: () -> Unit,
 ) {
-    when (conversation) {
-        is ConversationMissedCall -> CallConversationItem(
-            conversation,
-            eventType,
-            onConversationItemClick,
-            onConversationItemLongClick
-        )
-        is ConversationUnreadMention -> MentionConversationItem(
-            conversation,
-            eventType,
-            onConversationItemClick,
-            onConversationItemLongClick
-        )
-        is GeneralConversation -> GeneralConversationItem(
-            conversation,
-            eventType,
-            onConversationItemClick,
-            onConversationItemLongClick
-        )
-    }
+    MainConversationItem(
```

Comment:
minor thing :P but what is Main in this Context ?

Reply:
Renamed to GeneralConversationItem. It was a temporary name, because `General` was used elsewhere 😅